modules/base/rendering/renderableswitch.cpp

```
@@ -0,0 +1,141 @@
+#include <fstream>
+#include <optional>
+
+ //renderableSelectRenderable
+ //add by distance to the end
+ //renderableSelectorByDistance
+ //RenderableSwitcherByDistance
+namespace {
+    constexpr openspace::properties::Property::PropertyInfo AutoScaleInfo = {
+        "AutoScale",
+        "Auto Scale",
+        "When true, the plane will automatically adjust in size to match the aspect "
+        "ratio of the content. Otherwise it will remain in the given size."
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo DistanceThresholdInfo = {
+        "DistanceThreshold",
+        "Distance Threshold",
+        "Threshold for when the switch happens between the two renderables. "
+    };
+
+    struct [[codegen::Dictionary(RenderableSwitch)]] Parameters {
+        ghoul::Dictionary Renderable1;
+        ghoul::Dictionary Renderable2;
+
+        std::optional<float> distanceThreshold;
+    };
+#include "renderableswitch_codegen.cpp"
+} // namespace
+
+namespace openspace {
+
+    documentation::Documentation RenderableSwitch::Documentation() {
+        return codegen::doc<Parameters>(
+            "base_renderable_switch"
+        );
+    }
+
+    RenderableSwitch::RenderableSwitch(const ghoul::Dictionary& dictionary)
+        : Renderable(dictionary), _autoScale(AutoScaleInfo, false)
+        , _distanceThreshold(DistanceThresholdInfo, 1000000.f)
+    {
+        const Parameters p = codegen::bake<Parameters>(dictionary);
+
+        _renderable1 = createFromDictionary(p.Renderable1);
+
+        _renderable2 = createFromDictionary(p.Renderable2);
+
+        _distanceThreshold = p.distanceThreshold.value_or(_distanceThreshold);
+        addProperty(_distanceThreshold);
+
+        addProperty(_autoScale);
+    }
+
+    void RenderableSwitch::initializeGL() {
+        if (_renderable1) {
+            _renderable1->initializeGL();
+        }
+        if (_renderable2) {
+            _renderable2->initializeGL();
+        }
```

```suggestion
       ghoul_assert(_renderable1, "No renderable1");
       ghoul_assert(_renderable2, "No renderable1");

        _renderable1->initializeGL();
        _renderable2->initializeGL();
```

This should be enough.  The constructor will throw if it fails to create the renderables meaning that the RenderableSwitch will never be successfully created if both renderables are not valid.  Asserting here instead of hiding the error through the `if` statement better reflects the intended behavior